lib/membrane/testing/pipeline.ex

```
@@ -162,11 +163,31 @@ defmodule Membrane.Testing.Pipeline do
   end
 
   def start_link(pipeline_options, process_options \\ []) do
-    Pipeline.start_link(__MODULE__, default_options(pipeline_options), process_options)
+    do_start(:start_link, pipeline_options, process_options)
   end
 
   def start(pipeline_options, process_options \\ []) do
-    Pipeline.start(__MODULE__, default_options(pipeline_options), process_options)
+    do_start(:start, pipeline_options, process_options)
+  end
+
+  defp do_start(_type, %Options{elements: nil, module: nil}, _process_options),
+    do: {:error, :no_config}
```

This name

---

The solution for your naming problems is simple - don't use them at all. Instead of returning an error tuple, raise!